src/main/context-handler.ts

```diff
@@ -72,18 +72,34 @@ export class ContextHandler {
       this.prometheusProvider = service.id;
     }
 
-    return prometheusProviders.find(p => p.id === this.prometheusProvider);
+    return PrometheusProviderRegistry.getInstance().getByKind(this.prometheusProvider);
   }
 
-  async getPrometheusService(): Promise<PrometheusService | void> {
-    const providers = this.prometheusProvider ? prometheusProviders.filter(provider => provider.id == this.prometheusProvider) : prometheusProviders;
-    const prometheusPromises: Promise<PrometheusService | void>[] = providers.map(async provider => {
-      const apiClient = (await this.cluster.getProxyKubeconfig()).makeApiClient(CoreV1Api);
+  protected listPotentialProviders(): PrometheusProvider[] {
+    const registry = PrometheusProviderRegistry.getInstance();
 
-      return provider.getPrometheusService(apiClient);
-    });
+    if (typeof this.prometheusProvider === "string") {
+      return [registry.getByKind(this.prometheusProvider)];
+    }
+
+    return Array.from(registry.providers.values());
+  }
+
+  async getPrometheusService(): Promise<PrometheusService | undefined> {
+    const providers = this.listPotentialProviders();
+    const proxyConfig = await this.cluster.getProxyKubeconfig();
+    const apiClient = proxyConfig.makeApiClient(CoreV1Api);
+    const potentialServices = await Promise.allSettled(
```

Comment:
Does this alter priority of potential services? (test that confirms would be nice)

Reply:
It does not, it just means that if one of the `provider.getPrometheusService()` calls rejects, we can still proceed (namely it is skipped).

Reply:
Test added.